R/output_combined_table.R

```diff
@@ -0,0 +1,134 @@
+#'
+#' @return
+#' @export
+#'
+#' @examples
+database_create_combined_table <- function(database) {
+
+  location_latlon <-
+    database$locations %>%
+      dplyr::filter(location_property %in% c("latitude (deg)", "longitude (deg)")) %>%
+      tidyr::pivot_wider(names_from = location_property, values_from = value)
+
+  location_properties <-
+    database$locations %>%
+      dplyr::filter(!location_property %in% c("latitude (deg)", "longitude (deg)")) %>%
+      dplyr::mutate(
+        location_property = stringr::str_replace_all(location_property, "=", "-"),
+        value = stringr::str_replace_all(value, "=", "-"),
+        location_property = stringr::str_replace_all(location_property, ";", ","),
+        value = stringr::str_replace_all(value, ";", ",")
+      ) %>%
+      dplyr::mutate(location_properties = paste0(location_property, "=", value)) %>%
+      dplyr::select(dplyr::all_of(c("dataset_id", "location_id", "location_name", "location_properties"))) %>%
+      dplyr::group_by(dataset_id, location_id, location_name) %>%
+      dplyr::mutate(location_properties = paste0(location_properties, collapse = "; ")) %>%
+      dplyr::ungroup() %>%
+      dplyr::distinct()
+
+  contributors <-
+    database$contributors %>%
+      dplyr::mutate(
+        affiliation = stringr::str_replace_all(affiliation, ":", "-"),
+        affiliation = stringr::str_replace_all(affiliation, ";", ","),
+        affiliation = stringr::str_replace_all(affiliation, "<", "("),
+        affiliation = stringr::str_replace_all(affiliation, ">", ")"),
+        additional_role = stringr::str_replace_all(additional_role, "<", "("),
+        additional_role = stringr::str_replace_all(additional_role, ">", ")"),
+        data_collectors = paste0(given_name, " ", last_name),
+        data_collectors = ifelse(
+          !is.na(ORCID),
+          paste0(data_collectors, " <ORCID:", ORCID),
+          data_collectors),
+        data_collectors = ifelse(
+          is.na(ORCID),
+          paste0(data_collectors, " <affiliation:", affiliation),
+          paste0(data_collectors, ";affiliation:", affiliation)),
+        data_collectors = ifelse(
+          !is.na(additional_role),
+          paste0(data_collectors, ";additional_role:", additional_role, ">"),
+          paste0(data_collectors, ">"))
+      ) %>%
+      dplyr::select(-dplyr::all_of(c("last_name", "given_name", "ORCID", "affiliation", "additional_role"))) %>%
+      dplyr::group_by(dataset_id) %>%
+      dplyr::mutate(data_collectors = paste0(data_collectors, collapse = "; ")) %>%
+      dplyr::ungroup() %>%
+      dplyr::distinct()
+
+  contexts_tmp <-
+    database$contexts %>%
+      dplyr::mutate(
+        context_property = stringr::str_replace_all(context_property, "=", "-"),
+        value = stringr::str_replace_all(value, "=", "-"),
+        description = stringr::str_replace_all(description, "=", "-"),
+        context_property = stringr::str_replace_all(context_property, ";", ","),
```

Comment:
Do the characters "<" and ">" also need to be replaced in case they ever make it into the `context_property`, `value` and `description` fields?